os/kernel/sched/sched_is_sigwait.c

```diff
@@ -0,0 +1,92 @@
+/****************************************************************************
+ *
+ * Copyright 2023 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/************************************************************************
+ * Included Files
+ ************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <errno.h>
+#include <sched.h>
+
+/************************************************************************
+ * Definitions
+ ************************************************************************/
+
+/************************************************************************
+ * Private Type Declarations
+ ************************************************************************/
+
+/************************************************************************
+ * Global Variables
+ ************************************************************************/
+
+/************************************************************************
+ * Private Variables
+ ************************************************************************/
+
+/************************************************************************
+ * Private Function Prototypes
+ ************************************************************************/
+
+/************************************************************************
+ * Private Functionss
+ ************************************************************************/
+
+/************************************************************************
+ * Public Functions
+ ************************************************************************/
+
+/************************************************************************
+ * Name:  sched_is_sigwait
+ *
+ * Description:
+ *   This function returns whether the thread state is TSTATE_WAIT_SIG.
+ *
+ * Inputs
+ *   pid - the task ID of the task to query.
+ *
+ * Return Value:
+ *   On success, sched_is_sigwait() returns  1 or 0. On error, less than 
+ *   zero(0>) is returned.
+ * 
+ *   1  Thread state is TSTATE_WAIT_SIG.
+ *   0  Thread state is not TSTATE_WAIT_SIG.
+ *   -ESRCH  The task whose ID is pid could not be found.
+ *
+ ************************************************************************/
+
+int sched_is_sigwait(pid_t pid){
```

Comment:
This function seems very specific, what do you think about adding this code in work_signal itself? or changing the function to something like sched_get_task_state?
I see some similar things are being done directly in wqueu, so directly adding the code in work_signal should be fine I guess.

https://github.com/Samsung/TizenRT/blob/8491e79f3cf2dee2c39f983c00e51fc2dceb542d/os/wqueue/kwqueue/kwork_inherit.c#L92-L105

Reply:
Also, I think it's better to add the function in to os/kernel/task instead of sched.

Reply:
Thank you, I applied your review.
I added task_getstate() and modified to compare in kwork_signal().